storm-core/src/native/worker-launcher/impl/oci/oci_launch_cmd.c

```diff
@@ -0,0 +1,544 @@
+      free_olc_layers(layers, layer_index);
+      return NULL;
+    }
+
+    if (!parse_oci_launch_cmd_layer(&layers[layer_index], e)) {
+      free_olc_layers(layers, layer_index);
+      return NULL;
+    }
+
+    ++layer_index;
+  }
+
+  *num_layers_out = layer_index;
+  return layers;
+}
+
+static int parse_reap_layers_keep(cJSON* json) {
+  if (!cJSON_IsNumber(json)) {
+    fputs("ERROR: Bad/Missing OCI reap layer keep number\n", ERRORFILE);
+    return -1;
+  }
+  return json->valueint;
+}
+
+static void parse_oci_launch_cmd_oci_config(oci_config* oc, cJSON* oc_json) {
+  if (!cJSON_IsObject(oc_json)) {
+    fputs("ERROR: Bad/Missing OCI runtime config in launch command\n", ERRORFILE);
+    return;
+  }
+  oc->hostname = cJSON_DetachItemFromObjectCaseSensitive(oc_json, "hostname");
+  oc->linux_config = cJSON_DetachItemFromObjectCaseSensitive(oc_json, "linux");
+  oc->mounts = cJSON_DetachItemFromObjectCaseSensitive(oc_json, "mounts");
+
+  cJSON* process_json = cJSON_GetObjectItemCaseSensitive(oc_json, "process");
+  if (!cJSON_IsObject(process_json)) {
+    fputs("ERROR: Bad/Missing process section in OCI config\n", ERRORFILE);
+    return;
+  }
+  oc->process.args = cJSON_DetachItemFromObjectCaseSensitive(
+      process_json, "args");
+  oc->process.cwd = cJSON_DetachItemFromObjectCaseSensitive(
+      process_json, "cwd");
+  oc->process.env = cJSON_DetachItemFromObjectCaseSensitive(
+      process_json, "env");
+}
+
+static bool is_valid_layer_media_type(char* media_type) {
+  if (media_type == NULL) {
+    return false;
+  }
+
+  if (strcmp(SQUASHFS_MEDIA_TYPE, media_type)) {
+    fprintf(ERRORFILE, "ERROR: Unrecognized layer media type: %s\n", media_type);
+    return false;
+  }
+
+  return true;
+}
+
+static bool is_valid_oci_launch_cmd_layers(olc_layer_spec* layers,
+    unsigned int num_layers) {
+  if (layers == NULL) {
+    return false;
+  }
+  unsigned int i;
+  for (i = 0; i < num_layers; ++i) {
+    if (!is_valid_layer_media_type(layers[i].media_type)) {
+      return false;
+    }
+    if (layers[i].path == NULL) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+static bool is_valid_oci_config_linux_resources(const cJSON* oclr) {
+  if (!cJSON_IsObject(oclr)) {
+    fputs("ERROR: OCI config linux resources missing or not an object\n", ERRORFILE);
+    return false;
+  }
+
+  bool all_sections_ok = true;
+  const cJSON* e;
+  cJSON_ArrayForEach(e, oclr) {
+    if (strcmp("blockIO", e->string) == 0) {
+      // block I/O settings allowed
+    } else if (strcmp("cpu", e->string) == 0) {
+      // cpu settings allowed
+    } else if (strcmp("memory", e->string) == 0) {
+      // memory settings allowed. (added for storm. hadoop doesn't allow this)
+    } else {
+      fprintf(ERRORFILE,
+          "ERROR: Unrecognized OCI config linux resources element: %s\n", e->string);
+      all_sections_ok = false;
+    }
+  }
+
+  return all_sections_ok;
+}
+
+static bool is_valid_oci_config_linux_seccomp(const cJSON* ocls) {
```

Comment:
This JSON validation function is not implemented. Can it be similar to is_valid_oci_config_linux() ?

Reply:
secomp file is a little more involved so leaving it for future.